src/coroutine.rs

```diff
@@ -135,3 +158,21 @@ impl Coroutine {
         self.poll(py, None)
     }
 }
+
+impl Drop for Coroutine {
+    fn drop(&mut self) {
+        if self.future.is_some() {
+            Python::with_gil(|gil| {
+                let qualname = self
+                    .qualname
+                    .as_ref()
+                    .map_or(Ok("<coroutine>"), |n| n.as_ref(gil).to_str())
+                    .unwrap();
+                let message = format!("coroutine {} was never awaited", qualname);
```

Comment:
Does this only happen if it was never awaited or also if it was not polled to completion?

Reply:
In fact, instead of testing `self.future.is_some()`, I should test `self.waker.is_none()`, as it's always `Some(_)` when the coroutine has been polled at least once.